src/Compilers/Core/Portable/CommandLine/EditorConfig.SectionNameMatching.cs

```diff
@@ -0,0 +1,155 @@
+                        // Matches just this character
+                        sb.Append(lexer.EatCurrentCharacter());
+                        break;
+                    case TokenKind.Question:
+                        // '?' matches any single character
+                        sb.Append('.');
+                        break;
+                    case TokenKind.Star:
+                        // Matches any string of characters except directory separator
+                        // Directory separator is defined in editorconfig spec as '/'
+                        sb.Append("[^/]*");
+                        break;
+                    case TokenKind.StarStar:
+                        // Matches any string of characters
+                        sb.Append(".*");
+                        break;
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(tokenKind);
+                }
+            }
+            sb.Append('$');
+
+            return sb.ToString();
+        }
+
+        private ref struct SectionNameLexer
+        {
+            private readonly string _sectionName;
+
+            private int _position;
+
+            public SectionNameLexer(string sectionName)
+            {
+                _sectionName = sectionName;
+                _position = 0;
+            }
+
+            public bool IsDone => _position >= _sectionName.Length;
+
+            public TokenKind Lex()
+            {
+                int lexemeStart = _position;
+                switch (_sectionName[_position])
+                {
+                    case '*':
+                    {
+                        int nextPos = _position + 1;
+                        if (nextPos < _sectionName.Length &&
+                            _sectionName[nextPos] == '*')
+                        {
+                            _position += 2;
+                            return TokenKind.StarStar;
+                        }
+                        else
+                        {
+                            _position++;
+                            return TokenKind.Star;
+                        }
+                    }
+
+                    case '?':
+                        _position++;
+                        return TokenKind.Question;
+
+                    case '\\':
+                    {
+                        // Backslash escapes the next character
+                        int nextPos = _position + 1;
+                        if (nextPos >= _sectionName.Length)
+                        {
+                            return TokenKind.BadToken;
+                        }
+
+                        // Check for all of the possible escapes
+                        switch (_sectionName[nextPos])
+                        {
+                            case '\\':
+                                // "\\" -> "\"
+                                _position += 2;
+                                return TokenKind.Backslash;
```

Comment:
Not handled by the parser?

Reply:
Nice catch. And I clearly don't have a test, either. Will fix.